### DISCUSSION TITLE
714

### DISCUSSION THREAD

**Discussion by giscus** [jojoldu/blog-comments#896] [Comments]
# 714

Node.js 환경에서는 다수의 비동기 작업을 처리하기에 적합하다. 예를 들어, 다음과 같은 상황이 있다고 가정해보자. ID 별로 조회만 가능한 외부 서비스에서 대량의 건수를 API로 조회해야하는 경우 join, subquery 로 조회시 성능과 DB의 부하가 심하여, id 별로 쿼리를 나눠서 조회해야하는 경우 대량으로 이벤트 메세지를 발행해야하는 경우 등등 다량의 Promise 작업들은 백엔드 애플리케이션 작업에서는 언제든지 발생할 수 있다. 이럴때 Node.js 환경에서는 Promise.all 을 통해 쉽게 해결할 수 있다. 여기서 Promise.all 보다 좀 더 성능을 개선할 수 있는 방법을 보자. 개요 수백개 이상의 Promise 작업들은 보통 한번에 Promise.all을 수행하지 않는다. 예..

https://jojoldu.tistory.com/714

**Comment by mystyle2006** [comment]
좋은 라이브러리 하나 알아갑니다!! :) 감사합니다~

**Comment by backend-joonyoung** [comment]
chunk 단위는 50으로 설정했다.
measurePromiseAll 에서는 Chunk단위를 50개로 Promise 배열을 나누어 총 20개의 배열을 순차적으로 수행한다.
measurePromiseAll 에서는 50개의 레일을 깔아 Promise 1000개를 수행한다.

에서 measurePromiseAll-> measurePromisePool 로 수정되어야 하는것 같습니다 :)

**Reply by jojoldu** [reply]
아고 ㅋㅋ 오타 제보 감사합니다!

**Comment by gyunseo** [comment]
와 진짜 헷갈리는 개념이었는데, 잘 읽고 갑니다!
